edb/server/pgcon/pgcon.pyx

```diff
@@ -1096,10 +1213,9 @@ cdef class PGConnection:
                     if mtype == b'D':
                         # DataRow
                         if not has_result:
-                            raise errors.InternalServerError(
-                                f'query that was inferred to have '
-                                f'no data returned received a DATA package; '
-                                f'query: {query.sql}')
+                            # It's likely output_format=NONE, so just discard
```

Comment:
Can we pass output_format in the query unit to avoid guessing?

Reply:
Yup that's better - and makes script logic easier.